feat(agglayer): make bridge network ID a deployment setting (#3062)
0xMiden/protocol: CHANGELOG.md

# Changelog

## v0.16.0 (TBD)

> - [BREAKING] Changed `AggLayerBridge::new`, `create_bridge_account`, and `create_existing_bridge_account` to accept a `network_id: u32` parameter; the bridge now stores its AggLayer network ID in account storage and reads it at runtime instead of compiling in a hardcoded constant, so testnet and mainnet deployments share the same code commitment ([#3062](https://github.com/0xMiden/protocol/pull/3062)).

Author: ```suggestion
- [BREAKING] Changed AggLayerBridge to store its AggLayer network ID in account storage ([#3062](https://github.com/0xMiden/protocol/pull/3062)).
```
